infrastructure/modules/firewall/main.tf

```diff
@@ -0,0 +1,126 @@
+resource "aws_wafv2_web_acl" "waf_acl" {
+  name        = "${var.waf_name}${local.workspace_suffix}"
+  description = "CM Frontend WAF"
+  scope       = "CLOUDFRONT"
+
+  default_action {
+    allow {}
+  }
+
+  # Primary Web ACL metric
+  visibility_config {
+    cloudwatch_metrics_enabled = true
+    metric_name                = "${var.waf_acl_metric_name}${local.workspace_suffix}"
+    sampled_requests_enabled   = true
+  }
+
+  rule {
+    name     = "${var.waf_reputation_rule_name}${local.workspace_suffix}"
+    priority = 30
+    # action {
+    #   block {}
+    # }
+    override_action {
+      none {}
+    }
+
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesAmazonIpReputationList"
+        vendor_name = "AWS"
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "${var.ip_reputation_list_metric_name}${local.workspace_suffix}"
+      sampled_requests_enabled   = true
+    }
+  }
+
+  # Service-team specific rules
+  rule {
+    name     = "${var.non_gb_rule_metric_name}${local.workspace_suffix}"
+    priority = 1
+    action {
+      block {}
+    }
+    statement {
+      not_statement {
+        statement {
+          geo_match_statement {
+            country_codes = ["GB"]
+          }
+        }
+      }
+    }
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = var.non_gb_rule_metric_name
+      sampled_requests_enabled   = true
+    }
+  }
+
+}
+
```

Comment:
I think we'll need a rule for rate-limiting too. I had thought this came free out of the box, but I don't think it does! 
It should be simple enough to pop in. This should do the trick:

`{
  "Name": "nhse-uec-cm-custom-rate-limit-prototype",
  "Priority": 2,
  "Action": {
    "Block": {}
  },
  "VisibilityConfig": {
    "SampledRequestsEnabled": true,
    "CloudWatchMetricsEnabled": true,
    "MetricName": "nhse-uec-cm-custom-rate-limit-prototype"
  },
  "Statement": {
    "RateBasedStatement": {
      "Limit": 10000,
      "EvaluationWindowSec": 300,
      "AggregateKeyType": "IP"
    }
  }
}`

Might just want to parameterise the Name, Limit, and Evaluation Window...